### ISSUE TITLE
Mails not sent to OnPremise Destinations with port 465

### ISSUE THREAD

**Issue by mschfer** [SAP/cloud-sdk-js#5124]
**Describe the bug**
Since version `3.22.2` of Cloud SDK, mails cannot be transmitted any more via the cloud connector/OnPremise destination to an SMTP server running on port 465 (implicit TLS).

**To Reproduce**

1. Set up Cloud Connector Connection to SMTP Server on Port 465 (implicit TLS) and corresponding OnPremise destination `smtp`
2. call `sendMail()`

Now `sendMail()` does not resolve. Debugging showed that it is blocked by `await retrieveGreeting(socket)` in https://github.com/SAP/cloud-sdk-js/blob/b9d31bb4e85b68c6d9c3c802d5275a5d52c15658/packages/mail-client/src/mail-client.ts#L281 which does not resolve.

**Used Versions:**

- node version: 20.15.1
- npm version: 10.7.0
- SAP Cloud SDK version: 3.22.2

**Code Examples**
```js
sendMail(
    {'destinationName': 'smtp'},
   [{
      to: [{ address: 'user@example.com', name: 'Max Mustermann' }],
      subject: 'subject',
      html: 'html'
    }],
   {
   'secure': true,
   'sdkOptions': {
     parallel: false
   }
 }
)
```

**Log file**
```
Waiting for SMTP greeting message...
```

**Impact / Priority**

No mails are sent. No exception is thrown. The mail sending fails silently.

Affected development phase: Production

Impact: Impaired

Timeline: Already live

**Comment by ZhongpinWang** [comment]
Hi @mschfer ,

Thanks for reaching out.

Did the mail-client work for you prior to `3.22.2` and with port `465`? Which node version did you use before? We had a fix implemented in this patch release because the client was not working for node 20. After publishing this version, it works as reported in SAP/cloud-sdk-js#5015 and our e2e setup also seems to confirm the fix.  

The tested working setup is to use port `587` and set `secure` to `false`. Could you please try it out?

**Comment by mschfer** [comment]
Hi @ZhongpinWang,
thanks for the fast answer. 

Yes the mail-client did work prior to 3.22.2 with port `465` and `secure: true` i.e. implicit TLS, but not with port `587`/ STARTTLS. We are using node 20. 
With version `3.22.2` sending mails via port `465` does not work anymore as outlined above. It seems that the changes regarding resending the greeting in `3.22.2` have broken the support for implicit TLS. 
Instead port `587` and `secure: false` works fine as mentioned by you since version `3.22.2`. We are using this as a temporary fallback for now.
However, using STARTTLS is not recommended anymore and implicit TLS using port 465 is preferred (see https://datatracker.ietf.org/doc/html/rfc8314).

**Comment by ZhongpinWang** [comment]
Hi @mschfer ,

It is surprising that it worked for you before with node 20 and port 465. We will check if we can reproduce this issue on our side. This could take some time though due to our current capacity. Thanks again for reporting this issue.

**Comment by ZhongpinWang** [comment]
Hi @mschfer,

Thanks again for reporting this issue. We have implemented a fix in #5150. This will be released in the next patch version. In the meantime, you can try to verify the fix by using the `canary` version.

Feel free to re-open this issue if the problem still persists.
